testbed/tests/syslog_integration_test.go

```diff
@@ -0,0 +1,196 @@
+			severityNumber: 17,
+			severityText:   "err",
+			attributes: map[string]interface{}{
+				"message":  "-",
+				"priority": int64(19),
+				"facility": int64(2),
+			},
+		},
+	}
+
+	complementaryTest(t, "rfc3164", expectedData)
+}
+
+func componentFactories(t *testing.T) otelcol.Factories {
+	factories, err := testbed.Components()
+	require.NoError(t, err)
+	return factories
+}
+
+func complementaryTest(t *testing.T, rfc string, expectedData []expectedDataType) {
+	// Prepare ports
+	port := testbed.GetAvailablePort(t)
+	inputPort := testbed.GetAvailablePort(t)
+
+	// Start SyslogDataReceiver
+	syslogReceiver := datareceivers.NewSyslogDataReceiver(rfc, port)
+	backend := testbed.NewMockBackend("mockbackend.log", syslogReceiver)
+	require.NoError(t, backend.Start())
+	backend.EnableRecording()
+
+	// Prepare and run collector
+	config := `
+receivers:
+  syslog/client:
+    protocol: %s
+    tcp:
+      listen_address: '127.0.0.1:%d'
+exporters:
+  syslog/client:
+    endpoint: 127.0.0.1
+    network: tcp
+    protocol: %s
+    port: %d
+    tls:
+      insecure: true
+service:
+  pipelines:
+    logs/client:
+      receivers:
+        - syslog/client
+      exporters:
+        - syslog/client`
+
+	collector := testbed.NewInProcessCollector(componentFactories(t))
+	_, err := collector.PrepareConfig(fmt.Sprintf(config, rfc, inputPort, rfc, port))
+
+	require.NoError(t, err)
+	err = collector.Start(testbed.StartParams{
+		Name: "Agent",
+	})
+	require.NoError(t, err)
+
+	// prepare data
+
+	message := ""
+	expectedAttributes := []map[string]interface{}{}
+	expectedLogs := plog.NewLogs()
+	rl := expectedLogs.ResourceLogs().AppendEmpty()
+	lrs := rl.ScopeLogs().AppendEmpty().LogRecords()
+
+	for _, e := range expectedData {
+		lr := lrs.AppendEmpty()
+		lr.Body().SetStr(e.message)
+		lr.SetSeverityNumber(e.severityNumber)
+		lr.SetSeverityText(e.severityText)
+		lr.SetTimestamp(e.timestamp)
+		expectedAttributes = append(expectedAttributes, e.attributes)
+		message += e.message + "\n"
+	}
+
+	// Prepare client
+	conn, err := net.Dial("tcp", fmt.Sprintf("127.0.0.1:%d", inputPort))
+	require.NoError(t, err)
+
+	// Write requests
+	fmt.Fprint(conn, message)
+
+	// Wait for all messages
+	for len(backend.ReceivedLogs) < 1 {
+		time.Sleep(100 * time.Millisecond)
+	}
+
+	require.Equal(t, len(backend.ReceivedLogs), 1)
+	require.Equal(t, backend.ReceivedLogs[0].ResourceLogs().Len(), 1)
+	require.Equal(t, backend.ReceivedLogs[0].ResourceLogs().At(0).ScopeLogs().Len(), 1)
+	require.Equal(t, backend.ReceivedLogs[0].ResourceLogs().At(0).ScopeLogs().At(0).LogRecords().Len(), len(expectedData))
+
+	// Clean received logs
+	attributes := []map[string]interface{}{}
+
+	lrs = backend.ReceivedLogs[0].ResourceLogs().At(0).ScopeLogs().At(0).LogRecords()
+	for i := 0; i < lrs.Len(); i++ {
+		lrs.At(i).SetObservedTimestamp(0)
+
+		attributes = append(attributes, lrs.At(i).Attributes().AsRaw())
+		lrs.At(i).Attributes().Clear()
+	}
```

Comment:
I should perform assertions here instead of changing the received objects. Will improve it